utils/pckgen

```diff
@@ -0,0 +1,559 @@
+        done
+        LAST_COMMIT_HASH=$(git -C "$SOURCES_DIR" log --pretty=format:%H | head -n 1)
+        LAST_ABBR_COMMIT_HASH=$(git -C "$SOURCES_DIR" log --pretty=format:%h | head -n 1)
+        LAST_COMMIT_DATE=$(git -C "$SOURCES_DIR" log --pretty=format:%cI | head -n 1)
+        LAST_COMMIT_TIMESTAMP=$(git -C "$SOURCES_DIR" log --pretty=format:%ct | head -n 1)
+    elif [ -f "$VERSION_FILE" ]; then
+        source "$VERSION_FILE" &> /dev/null
+    else
+        error "Couldn't obtain CAS version - no git tree nor version file within given sources"
+    fi
+
+    VERSION_LONG="${CAS_VERSION_MAIN}.${CAS_VERSION_MAJOR}.${CAS_VERSION_MINOR}.${CAS_VERSION_BUILD}"
+
+    if [ $MASTER_PACKAGE ]; then
+        VERSION="$VERSION_LONG"
+        RELEASE="master"
+    elif [ $RELEASE_PACKAGE ]; then
+        VERSION="$VERSION_SHORT"
+        RELEASE="release"
+    else
+        VERSION="$VERSION_LONG"
+        RELEASE="devel.${LAST_COMMIT_TIMESTAMP}"
+    fi
+}
+
+create_version_file() {
+    echo "--- Creating version file"
+    if [ -f "$VERSION_FILE" ] && [ ! $FORCE ]; then
+        echo "---   Version file already exists (use --force to overwrite)"
+    else
+        mkdir -p $(dirname "$VERSION_FILE")
+        touch "$VERSION_FILE"
+        if [ $? -ne 0 ]; then
+            error "Couldn't create version file"
+        fi
+
+        CREATION_DATE=$(date --iso-8601=seconds)
+        CREATION_TIMESTAMP=$(date +%s)
+        echo "CREATION_DATE=$CREATION_DATE" > "$VERSION_FILE"
+        echo "CREATION_TIMESTAMP=$CREATION_TIMESTAMP" >> "$VERSION_FILE"
+        echo "CAS_VERSION_MAIN=$CAS_VERSION_MAIN" >> "$VERSION_FILE"
+        echo "CAS_VERSION_MAJOR=$CAS_VERSION_MAJOR" >> "$VERSION_FILE"
+        echo "CAS_VERSION_MINOR=$CAS_VERSION_MINOR" >> "$VERSION_FILE"
+        echo "CAS_VERSION_BUILD=$CAS_VERSION_BUILD" >> "$VERSION_FILE"
+        echo "LAST_COMMIT_HASH=$LAST_COMMIT_HASH" >> "$VERSION_FILE"
+        echo "LAST_ABBR_COMMIT_HASH=$LAST_ABBR_COMMIT_HASH" >> "$VERSION_FILE"
+        echo "LAST_COMMIT_DATE=$LAST_COMMIT_DATE" >> "$VERSION_FILE"
+        echo "LAST_COMMIT_TIMESTAMP=$LAST_COMMIT_TIMESTAMP" >> "$VERSION_FILE"
+    fi
+}
+
+archive_prepare() {
+    echo "--- Preparing sources to archive"
+    if [ $ARCHIVE_PREPARED ] && [ ! $FORCE ]; then
+        echo "---   Sources already prepared (use --force to overwrite)"
+    else
+        echo "--- Making copy to work on"
+        local TEMP_SOURCES_DIR="$TEMP_DIR/$CAS_FILENAME"
+        rm -rf "$TEMP_SOURCES_DIR"
+        mkdir -p "$TEMP_SOURCES_DIR"
+        cp -fat "$TEMP_SOURCES_DIR" "$SOURCES_DIR/."
+
+        if [ -d "$TEMP_SOURCES_DIR/.git" ]; then
+            echo "--- Updating submodules"
+            for SUBMOD in ${SUBMODULES[@]}; do
+                if [ ! -f "$TEMP_SOURCES_DIR/$SUBMOD/.git" ]; then
+                    local SUBMODULES_UPDATE=1
+                fi
+            done
+            if [ ! $SUBMODULES_UPDATE ] && [ ! $FORCE ]; then
```

Comment:
I'm not sure whether I get this right.

If you run the script without --force then :
a) if all modules are initialized, you are going to print a warning "Submodules already initialized (use --force to update)"
b) if some modules are not initialized, you don't print anything, intialize the ones that are missing and update all of them silently.

This seems inconsistent